### PR TITLE
增加drop table删除表缓存支持

### PR DESCRIPTION
**Pull request by jilieryuyi**
增加了删除表清理缓存的支持
对应issues
https://github.com/siddontang/go-mysql/issues/199

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @jilieryuyi 

The feature you add is in the checkRenameTable function, this is not suitable. You should use another function to handle it.

**Comment by siddontang** [comment]
you also need to add some tests for this.

**Comment by jilieryuyi** [comment]
@siddontang ok

**Comment by jilieryuyi** [comment]
test case:
ALTER TABLE `dt_decimal_1` CHANGE `c` `c` DECIMAL(38,1) NULL DEFAULT NULL;
drop table dt_float_1;

debug info：
INFO[2017-12-09 07:56:21] table structure changed, clear table cache: test.dt_decimal_1
 
DEBU[2017-12-09 07:56:21] binlog事件：OnDDL                               
DEBU[2017-12-09 07:56:21] update master position (mysql-bin.000049, 2098) 
DEBU[2017-12-09 07:56:21] binlog事件：OnPosSynced (mysql-bin.000049, 2098) %!b(bool=true) 
INFO[2017-12-09 07:56:51] table structure changed, clear table cache: test.dt_float_1
 
DEBU[2017-12-09 07:56:51] binlog事件：OnDDL                               
DEBU[2017-12-09 07:56:51] update master position (mysql-bin.000049, 2286) 
DEBU[2017-12-09 07:56:51] binlog事件：OnPosSynced (mysql-bin.000049, 2286) %!b(bool=true)

**Comment by jilieryuyi** [comment]
rename bug : rename should clear old table cache
It has been fixed

test case:
RENAME TABLE dt_decimal_25 TO dt_decimal_26;

debug info:
 INFO[2017-12-09 08:21:15] table structure changed, clear table cache: test.dt_decimal_25 
 
DEBU[2017-12-09 08:21:15] binlog事件：OnDDL                               
DEBU[2017-12-09 08:21:15] update master position (mysql-bin.000049, 3559) 
DEBU[2017-12-09 08:21:15] binlog事件：OnPosSynced (mysql-bin.000049, 3559) %!b(bool=true)

File: canal/canal_test.go [hunk 1]
```diff
@@ -171,3 +171,23 @@ func TestRenameTableExp(t *testing.T) {
...
+		"drop table `test1`",
+		"DROP TABLE `test1`",
+		"DROP table `test`.`test1` IF EXISTS `test`.`test1`",
+		"DROP TABLE `test1` /* generated by server */",
+	}
+
+	table := []byte("test1")
+	for _, s := range cases {
+		m := expDropTable.FindSubmatch([]byte(s))
+		if m == nil || !bytes.Equal(m[2], table) {
```
Comment (siddontang) [inline canal/canal_test.go]: seem m here cannot be nil?

File: canal/canal_test.go [hunk 2]
```diff
@@ -157,12 +157,12 @@ func TestRenameTableExp(t *testing.T) {
 		"rename table mydb.mytable to mydb.mytable1",
 		"rename table mytable to mytable1",
 
-		"rename table `mydb`.`mytable1` to `mydb`.`mytable2`, `mydb`.`mytable` to `mydb`.`mytable1`",
-		"rename table `mytable1` to `mytable2`, `mytable` to `mytable1`",
-		"rename table mydb.mytable1 to mydb.mytable2, mydb.mytable to mydb.mytable1",
-		"rename table mytable1 to mytable2, mytable to mytable1",
+		"rename table `mydb`.`mytable` to `mydb`.`mytable2`, `mydb`.`mytable3` to `mydb`.`mytable1`",
```
Comment (siddontang) [inline canal/canal_test.go]: why changing these tests?

Comment (jilieryuyi) [reply canal/canal_test.go]: rename should clear old table cache,  	expRenameTable = regexp.MustCompile("(?i)^RENAME\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s{1,}TO\\s.*?")
this is change for get the old table

Comment (siddontang) [reply canal/canal_test.go]: got it

File: canal/canal_test.go (hunk 1 above)
Comment (siddontang) [reply canal/canal_test.go]: any update? if m is nil, is it wrong?

Comment (siddontang) [reply canal/canal_test.go]: ping @jilieryuyi

**Comment by siddontang** [comment]
any update @jilieryuyi ?

**Comment by jilieryuyi** [comment]
@siddontang 哥，我们说中文吧，不是很懂你这边表达什么意思。我这边的测试样例也写了，测试也ok，还缺什么 ？

**Comment by jilieryuyi** [comment]
@siddontang  it was updated！

File: canal/canal_test.go (hunk 1 above)
Comment (NearTan) [reply canal/canal_test.go]: I think here just for validating the regex expression, `m` could be `nil` when that is mismatch, here lgtm.
cc @jilieryuyi

**Pull request merged**